data-processing/pulsar-flink/2.7.0/pulsar-flink.md

```diff
@@ -0,0 +1,536 @@
+---
+description: Pulsar Flink Connector allows Flink to read data from Pulsar and write data to Pulsar
+author: ["StreamNative"]
+contributors: ["StreamNative"]
+language: Java
+document: 
+source: "https://github.com/streamnative/pulsar-flink/tree/release-2.7.0"
+license: Apache License 2.0
+license_link: "https://www.apache.org/licenses/LICENSE-2.0"
+tags: ["Pulsar", "Flink", "Connector"]
+alias: Pulsar Flink Connector
+features: ["Pulsar Flink Connector allows Flink to read data from Pulsar and to write data to Pulsar"]
+icon: "/images/data-processing/flink-logo.png"
+download: "https://bintray.com/streamnative/maven/io.streamnative.pulsar-flink/2.7.0"
+support: StreamNative
+support_link: https://streamnative.io
+support_img: "/images/streamnative.png"
+owner_name: "StreamNative"
+owner_img: "/images/streamnative.png"
+dockerfile: 
+id: "pulsar-flink"
+---
+
+Pulsar Flink Connector is an integration of [Apache Pulsar](https://pulsar.apache.org/en/) and [Apache Flink](https://flink.apache.org/) (data processing engine), which allows Flink to read data from Pulsar and write data to Pulsar and provides exactly-once source semantics and at-least-once sink semantics.
+
+![](/images/data-processing/pulsar-flink-connector.png)
+
+For more information about how to use Pulsar Flink Connector, see [here](https://streamnative.io/docs/v1.0.0/process/pulsar-flink-connector/link/).
+
+# Prerequisites
+
+- Java 8 or higher version
+- Flink 1.9.0 or higher version
+- Pulsar 2.4.0 or higher version
+
+# Basic information
+
+This section describes basic information about the Pulsar Flink connector.
+
+## Client
+
+Currently, the following Flink versions are supported.
+
+- Flink 1.9 - 1.10: they are maintained in the [`flink-1.9` branch](https://github.com/streamnative/pulsar-flink/tree/flink-1.9).
+
+- Flink 1.11: it is maintained in the [`flink-1.11` branch](https://github.com/streamnative/pulsar-flink/tree/flink-1.11).
+
+- Flink 1.12: it is maintained in the [`master` branch](https://github.com/streamnative/pulsar-flink/tree/master).
+
+> **Note**  
+> Since Flink's API has changed greatly, we mainly work on new features in the `master` branch and fix bugs in other branches.
```

Comment:
Avoid using first person, which is indicated by the personal pronouns `I` and `we`. The first person
focuses on the writer, not on the reader or the information being presented. The first person can
also contribute to an overly informal tone.

Examples (incorrect)
I find some servers more difficult to configure.
Let’s start by planning the home page for the website.
We can add a model to the project that we created in the previous step.

Examples (correct)
Some servers can be more difficult to configure.
Start by planning the home page for the website.
You can add a model to the project that you created in the previous step.

The use of first person is acceptable in these situations:
• In the question portion of frequently asked questions (FAQs)
• In articles, white papers, or documents that have listed authors and in which the authors describe their own actions or opinions

Examples (incorrect)
Q: How does one change the properties of a class?
Tests were conducted on servers in three different environments.

Examples (correct)
Q: How do I change the properties of a class?
We conducted tests on servers in three different environments.